src/worker.cc

```diff
@@ -1135,12 +1137,17 @@ worker_t::result_t jsonify_error(const valhalla_exception_t& exception,
 
   // overwrite with osrm error response
   if (request.options.format() == odin::DirectionsOptions::osrm) {
-    auto found = OSRM_ERRORS_CODES.find(exception.code);
-    if (found == OSRM_ERRORS_CODES.cend()) {
-      found = OSRM_ERRORS_CODES.find(199);
+    if (exception.code == 154) { // special case - return valhalla error.
+      body << (request.options.has_jsonp() ? request.options.jsonp() + "(" : "")
+           << R"({"code":"NoRoute","message":"Path distance exceeds the max distance limit."})"
+           << (request.options.has_jsonp() ? ")" : "");
+    } else {
+      auto found = OSRM_ERRORS_CODES.find(exception.code);
+      if (found == OSRM_ERRORS_CODES.cend())
+        found = OSRM_ERRORS_CODES.find(199);
+      body << (request.options.has_jsonp() ? request.options.jsonp() + "(" : "") << found->second
+           << (request.options.has_jsonp() ? ")" : "");
```

Comment:
instead of all of this, couldnt you just change the code above to be:

    {154, R"({"code":"NoRoute","message":"Path distance exceeds the max distance limit."})"},